packages/web-app-files/src/views/spaces/GenericSpace.vue

```diff
@@ -321,7 +323,23 @@ export default defineComponent({
     }
   },
 
-  mounted() {
+  async mounted() {
+    if (isPublicSpaceResource(this.space)) {
+      let publicSpace
+      try {
+        publicSpace = await this.$clientService.webdav.getFileInfo(this.space)
+      } catch (e) {
+        publicSpace = this.space
+      }
+
+      if (linkRoleUploaderFolder.bitmask(false) === publicSpace.publicLinkPermission) {
+        return this.$router.push({
+          name: locationPublicUpload.name,
+          params: { token: this.space.id }
+        })
+      }
+    }
```

Comment:
Not the biggest fan of putting such logic in the view directly... But the public link is already resolved here, so I don't see another good place to put it.

Reply:
What about putting it in the `DriveResolver.vue`? 🤔

Reply:
This would work as well I think, but do you think it's a better fit though? It's "just another view". Also, the logic only applies to the `GenericSpace` view, not to `GenericTrash` and `DriveRedirect`.

Reply:
I moved it to the DriveResolver. I agree it's a better fit than the already resolved space.